### PR TITLE
Small test for Image class.

### PR DESCRIPTION
**Pull request by r0b1n**
This tests green for current implementation. 
BUT, I am not sure is it ok to separate empty tag and default 'latest' tag. Maybe in case of empty tag we should use 'latest' as default value?

### CODE REVIEW

**Comment by hacfi** [comment]
@r0b1n The tests look good but I don’t understand what you want to change. Tag defaults to latest. In your tests it doesn’t because you pass an empty string and not null for $tag.

**Comment by r0b1n** [comment]
No matter in this case `null` or empty string.
Check it:
```php
$image1 = new Image('rep', '');
$image2 = new Image('rep', null);
$image3 = new Image('rep');

var_dump($image1->__toString());
var_dump($image2->__toString());
var_dump($image3->__toString());
```
Gives:
```php
string(3) "rep"
string(3) "rep"
string(10) "rep:latest"
```

Quite tricky...

**Comment by hacfi** [comment]
@r0b1n Ok. So is there anything you want to change about the current implementation?

File: src/Docker/Tests/ImageTest.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                'testuser/testimage',
+                'sometag',
+                'testuser/testimage:sometag',
+                'long_image_id_here'
+            ],
+
+            [
+                'testuser/testimage',
+                '', // empty tag
+                'testuser/testimage',
```
Comment (hacfi) [inline src/Docker/Tests/ImageTest.php]: Do you want this to return `testuser/testimage:latest`?

Comment (r0b1n) [inline src/Docker/Tests/ImageTest.php]: Yep, it is more intuitive like for me. 
If tag not explicitly specified or it is empty - set tag to `latest`.
But I am not quite familiar with project, maybe I am wrong about this and empty tag is ok. :)

Comment (hacfi) [inline src/Docker/Tests/ImageTest.php]: Now I get your point.

As the Docker API defaults to `latest` if no tag is provided I’d rather change the implementation to never return `:latest`.

**Comment by hacfi** [comment]
Let’s merge the tests and discuss it further in another issue (PRs will be easier with these tests on master).

**Comment by joelwurtz** [comment]
IMO null and empty value should be invalid, as a tag is necessary for an image version

**Comment by hacfi** [comment]
@joelwurtz I’m fine with the current implementation but maybe don’t return the tag if it’s `latest` when using __toString.

**Comment by ubermuda** [comment]
I think the question here is can we trust docker to always fallback the absence of a tag to `latest`, and even if we can, should we?

I like Joel's suggestion to make an empty tag invalid. This way, it's no harder to use `latest` (you just don't pass a tag argument and use the default), and it's also future-proof because you will always have an explicit tag.

TL;DR: +1 for making empty tag invalid

**Comment by ubermuda** [comment]
(and +1 for doing that in another PR too)

**Comment by ubermuda** [comment]
Oh and does anyone have any idea why travis is failing again?

**Comment by hacfi** [comment]
@ubermuda Nope but I will try to fix this.

**Comment by joelwurtz** [comment]
Since the new version there is no more check on the image name

IMO docker daemon will throw an error if image name is invalid, so don't think this is necessary anymore.
